sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -242,6 +244,12 @@ def wait_for_idle(self, seconds_to_wait=None):
                 self.logger.error(error)
                 raise LivyClientTimeoutException(error)
 
+            if constants.YARN_RESOURCE_LIMIT_MSG in self._logs and \
```

Comment:
Can we put in a feature request on the Livy side to add a special error code for this case?  I am fine with this in the short-term but it's brittle to put this logic in Sparkmagic.